controllers/mirrorpeer_controller.go

```diff
@@ -106,6 +108,30 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.V(2).Info("All validations for MirrorPeer passed", "MirrorPeer", req.NamespacedName)
 
+	mirrorPeerCopy := mirrorPeer.DeepCopy()
+	if mirrorPeerCopy.Labels == nil {
+		mirrorPeerCopy.Labels = make(map[string]string)
+	}
+
+	if val, ok := mirrorPeerCopy.Labels[hubRecoveryLabel]; !ok || val != "resource" {
```

Comment:
Can we add unit tests for these cases?

Reply:
Added a test for checking the availability of label and its value after a reconcile